docs/pipelines/overview/configuration.md

```diff
@@ -1,95 +1,368 @@
+
+The following is an example of a minimal configuration that would allow Pipelines to determine how to authenticate with AWS for exactly one unit of IAC:
+
+```hcl
+# gruntwork.hcl
+unit {
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = {
+        plan_iam_role = "role-to-assume-for-plans"
+        apply_iam_role = "role-to-assume-for-applies"
+      }
+    }
+  }
+}
+```
+
+Placing this configuration in a `gruntwork.hcl` file in the same directory as a `terragrunt.hcl` file will cause Pipelines to assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands, using [OIDC](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) to authenticate to AWS and assume that role.
+
+In most circumstances, the same role would be assumed by multiple units of configuration within a repository (e.g. all units within a given directory configure resources for the same AWS account). In that circumstance, it would be more appropriate not to define a `gruntwork.hcl` file at all, and instead define a `.gruntwork` directory at the root of the repository with a file ending in `.hcl` that contains configurations that are applicable to all units within that directory.
+
+e.g.
+
+```hcl
+# .gruntwork/environments.hcl
+environment "an_account" {
+  filter {
+    paths = ["an-account/*"]
+  }
+
+  authentication {
+    aws_oidc {
+      account_id = "an-aws-account-id"
+      authentication_profile = {
+        plan_iam_role = "role-to-assume-for-plans"
+        apply_iam_role = "role-to-assume-for-applies"
+      }
+    }
+  }
+}
+```
+
+In this example, all the units located within the `an-account` directory sibling to the `.gruntwork` directory will assume the `role-to-assume-for-plans` role in the AWS account with ID `an-aws-account-id` when running Terragrunt plan commands by Pipelines.
+
+A typical approach to building Pipelines configurations is to first define minimal configurations that address the most common use-cases, and then to refactor and generalize those configurations as needed to reduce repetition.
+
+More details on how these configurations are defined will be detailed below.
+
+## Global Configurations
+
+The configurations found within a `.gruntwork` directory found in the current working directory, or a parent directory of the current working directory are referred to as global configurations. These configurations are typically applicable within a wide range of contexts within a repository, and are the primary mechanism for configuring Pipelines.
+
+Pipelines will attempt to find exactly one directory named `.gruntwork` when it is attempting to discover configurations. It will not continue to search for configurations in parent directories once it finds a `.gruntwork` directory.
```

Comment:
This is a smidge complicated, maybe worth a visual or an example

Reply:
Deferring on this for now.